main_f4.c

```diff
@@ -331,6 +416,19 @@ main(void)
 	/* do board-specific initialisation */
 	board_init();
 
+	/* configure the clock for bootloader activity */
+	rcc_clock_setup_hse_3v3(&clock_setup);
```

Comment:
I have had to move the clock setup earlier in the code, so that I can setup the timer correctly to sample the USART RX pin in order to determine if a break is being received.